src/qiboml/models/encoding.py

```diff
@@ -62,7 +65,15 @@ def __call__(self, x: ndarray) -> Circuit:
                 f"Invalid input dimension {x.shape[-1]}, but the allocated qubits are {self.qubits}.",
             )
         circuit = self.circuit.copy()
-        ones = np.flatnonzero(x.ravel() == 1)
-        for bit in ones:
-            circuit.add(gates.X(self.qubits[bit]))
+
+        def true_fn():
+            circuit.add(gates.X(q))
+
+        def false_fn():
+            tf.no_op()
+
+        for i, q in enumerate(self.qubits):
+            pred = tf.equal(x[0][i], 1)
+            tf.cond(pred, true_fn=true_fn, false_fn=false_fn)
+
```

Comment:
Same here, the separation of true and false branches is fine, but you can't use `tf` functions. In any case, #55 is slightly changing the binary encoding and should lift the problems related to `tf` symbolic tensors here